### PR TITLE
do not return current date when timestamp is undefined

### LINKED ISSUE #3474: formatUnixTimestamp returns current date when timestamp is undefined
**Describe the bug**
`formatUnixTimestamp` returns the current date when the timestamp passed in is `undefined`.

While this is down to the default behaviour of the `moment` call it can result in cases where we show the incorrect timestamp if we don't plumb data to timestamp related components and functions correctly because we might not easily spot the difference between the current date and other dates in different languages.

**To Reproduce**
`formatUnixTimestamp(); // current date`

**Expected behaviour**
`formatUnixTimestamp(); // undefined`
`formatUnixTimestamp({ timestamp: Date.now() }); // current date`

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Testing notes**

- [ ] This bug fix is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by jroebu14**
Resolves https://github.com/bbc/psammead/issues/3474

**Overall change:**
Fixes bug where `formatUnixTimestamp` returns the current date when you do not pass in a timestamp.

This will help us more easily catch cases when a timestamp is undefined because instead of outputting today's date, which can be difficult to spot the difference in another language, it won't output anything.

**Code changes:**

- Early return if `timestamp` value is undefined
- Add tests to make this a requirement of the function

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Automated jest tests added (for new features) or updated (for existing features)
- [ ] This PR requires manual testing

### CODE REVIEW

**Review by ryanmccombe — APPROVED** [review]
I think the bug might be in our use of this module, rather than the module itself

Using the current time might be a sensible default.  It provides a quick way to output a formatted datestamp, just like the moment implementation

Either way is good though 😄 just needs a version bump

**Comment by jroebu14** [comment]
@ryanmccombe in that case you could do `formatUnixTimestamp({ timestamp: Date.now() })`

File: packages/containers/psammead-timestamp-container/CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 <!-- prettier-ignore -->
 | Version | Description |
 |---------|-------------|
+| 4.0.0 | [PR#3468](https://github.com/bbc/psammead/pull/3472) Do not return current date in formatUnixTimestamp when timestamp is undefined |
```
Comment (ryanmccombe) [inline packages/containers/psammead-timestamp-container/CHANGELOG.md]:
```suggestion
| 4.0.0 | [PR#3472](https://github.com/bbc/psammead/pull/3472) Do not return current date in formatUnixTimestamp when timestamp is undefined |
```

**Comment by ryanmccombe** [comment]
Just wanted to clarify because I think my post was confused given the standup update

I'm not saying we don't have a bug here, I'm just saying that having a date formatter use the current date by default is not necessarily where the bug is.

This is an established behavior in Moment, as well as JS itself (calling the date constructor with no arguments assumes you want the current date - `new Date()`)

Like you said, I'm fairly ambivalent _how_ we fix the bug - but I'm not ambivalent about _whether_ we fix the bug 😛

**Review by RichardPK — APPROVED** [review]
Good solution and important to help debugging imo. I think you mentioned you had - but worth getting @sareh 's 👀 across this?

**Comment by jroebu14** [comment]
> Nothing is better than incorrect information

 _Lily Higham_

**Pull request merged**